pkg/blockchain/blockchainPublisher.go

```diff
@@ -155,25 +157,83 @@ func (m *BlockchainPublisher) PublishGroupMessage(
 				return nil, errors.New("transaction receipt is nil")
 			}
 
-			return findLog(
+			return findLogs(
 				receipt,
 				m.messagesContract.ParseMessageSent,
-				"no message sent log found",
+				1,
+			)
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return logs[0], nil
+}
+
+func (m *BlockchainPublisher) BootstrapGroupMessages(
```

Comment:
With multiple messages it becomes easier for us to exceed the limits on a single tx. Are we planning some mechanism to break up multiple payloads into smaller chunks if we get errors there?

Reply:
I plan to use this function to bootstrap one message at a time. Initially I don't think we need to batch anything and risk reverting tx's.

The amount of messages to backfill can be easily migrated in a few days, and then the migrator would continue pulling these on the fly.

If down the road we're confident this can be used with batches then I'll modify it to guarantee the data published in the batch is not bigger than the max data a tx can hold.

Reply:
Glad I asked. That makes sense